### DISCUSSION TITLE
[SOLVED] switch esc (exit insert mode) and capslock

### DISCUSSION THREAD

**Discussion by aum7** [LunarVim/LunarVim#3796] [Q&A] [answered]
olo
big thanks to devs for making & sharing this collection
love it
has got everything i missed in ms code
and so much more

q : i would like to switch esc <-> capslock functionality
because capslock is closer for left pinky
and capslock is not really used, especially in my favorite - lua d' great

expected behavior : 
pressing *capslock* - exit insert mode
pressing *esc* - toggle capslock on / off (this is not mandatory)

please advise

long live lunarvim

have fun
aum
s love nia

**Comment by brneor** [comment]
Not sure, but I think you can only achieve this behaviour on a system-level configuration. This would heavily vary according to your distro if you're using linux, but you can achieve on some of them with

```bash
xmodmap -e "keycode 9 = Caps_Lock"; xmodmap -e "keycode 66 = Escape"
```

or 

```bash
setxkbmap -option caps:swapescape
```

On Windows you can use [AutoHotkey](https://www.autohotkey.com/).

**Comment by aum7** [comment]
@ brneor thank your for the answer, but i do not want this system-wide, though it did crossed my mind, for a shortest moment possible, lol

i just need it when using lvim / nvim

windows i am trying to get completely rid of, for over a decade now, and am finally making successful transition; 1st reason is - totally user unfriendly, 2nd reason - programmer unfriendly; there is more, but we'll stop here

**Comment by aum7** [comment, marked as answer]
after some reading on net, i got an alternative solution
i tried suggested solutions, and just added
lvim.keys.insert_mode["jk"] = "\<esc\>"
to my ~./config/lvim/config.lua file

so, in this case, while in insert mode, pressing jk will act as pressing esc key
and switch to normal mode

issue solved

**Reply by brneor** [reply]
I'm glad you found a solution! I suggested the change system wide because it's how I've been using for the past 5 years and have been great: in Caps Lock's place I have a key that acts like an Esc when tapped and like a Control when held. My Caps Lock is now activated and deactivated by pressing Left Shift twice.

**Reply by aum7** [reply]
out of curiosity, how would capslock be identified in code
i tried \<caps\>, but it did not work as expected
to clarify, escape key is identified as
\<esc\> in the lvim config.lua file
what is equivalent for capslock ?

long live l' lunarvim !
lol

have fun

**Reply by brneor** [reply]
I'm not sure, in my keyboard this is set at firmware level.
